### PR TITLE
Upgrade Rails

### PR DESCRIPTION
**Pull request by nemodjuric**
Related to 
https://github.com/department-of-veterans-affairs/caseflow-feedback/pull/171
https://github.com/department-of-veterans-affairs/caseflow-efolder/pull/939

### CODE REVIEW

File: app/controllers/issues_controller.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ def issue_params
               :issue,
               :level_1,
               :level_2,
-              :level_3)
+              :level_3).to_h
```
Comment (nemodjuric) [inline app/controllers/issues_controller.rb]: to_h fixed some of the tests and it's a result of http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#actioncontroller-parameters-no-longer-inherits-from-hashwithindifferentaccess

File: app/controllers/test/users_controller.rb [hunk 2]
```diff
@@ -89,7 +89,7 @@ def save_admin_login_attempt
   def set_end_products
     BGSService.end_product_records[:default] = new_default_end_products
 
-    render nothing: true, status: 200
+    head :ok
```
Comment (nemodjuric) [inline app/controllers/test/users_controller.rb]: :nothing` option is deprecated and removed in Rails 5.1

File: app/models/dispatch_stats.rb [hunk 3]
```diff
@@ -13,7 +13,7 @@ class DispatchStats < Caseflow::Stats
     end,
 
     establish_claim_active_users: lambda do |range|
-      EstablishClaim.where(completed_at: range).pluck(:user).uniq.count
```
Comment (nemodjuric) [inline app/models/dispatch_stats.rb]: Passing a class as a value in an Active Record query is deprecated.

File: app/models/task.rb [hunk 4]
```diff
@@ -86,7 +86,7 @@ def joins_task_result
     end
 
     def todays_quota
-      TeamQuota.find_or_create_by!(date: Time.zone.today, task_type: self)
```
Comment (nemodjuric) [inline app/models/task.rb]: Passing a class as a value in an Active Record query is deprecated

File: spec/services/appeals_repository_spec.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@
     @old_repo = Appeal.repository
     Appeal.repository = AppealRepository
 
-    allow_any_instance_of(VACOLS::Case::ActiveRecord_Relation).to receive(:find).and_return(nil)
```
Comment (nemodjuric) [inline spec/services/appeals_repository_spec.rb]: Model ActiveRecord Relation was set as private constant
https://github.com/rails/rails/issues/30943

File: app/models/annotation.rb [hunk 6]
```diff
@@ -1,4 +1,4 @@
-class Annotation < ActiveRecord::Base
+class Annotation < ApplicationRecord
```
Comment (nemodjuric) [inline app/models/annotation.rb]: http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#active-record-models-now-inherit-from-applicationrecord-by-default

**Comment by sansb** [comment]
This is blowing up on Circle, seems like the same error every time-
```      
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "users" does not exist
```

Any idea what might be causing this?

**Comment by nemodjuric** [comment]
@sansb I believe this is the reason: https://circleci.com/docs/1.0/ruby-exception-during-schema-load/

**Pull request merged**